test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -688,7 +689,12 @@ public static void assertVersionSerializable(Version version, Streamable streama
                 input = new NamedWriteableAwareStreamInput(input, namedWriteableRegistry);
             }
             input.setVersion(version);
-            newInstance.readFrom(input);
+            // This is here since some Streamables are being converted into Writeables
+            // and the readFrom method throws an exception if called
```

Comment:
So instead you first try and deserialize from our normal ctor. Yeah, this makes sense and the way you have it rigged looks about right for a temporary measure.